### PR TITLE
Fix for halium 7.1 camera and other related issues

### PR DESCRIPTION
**Pull request by rubencarneiro**

### CODE REVIEW

File: ubuntu/70-hammerhead.rules [hunk 1]
```diff
@@ -108,9 +108,16 @@ ACTION=="add", KERNEL=="qmi0", OWNER="radio", GROUP="radio", MODE="0640"
...
 ACTION=="add", KERNEL=="qmi2", OWNER="radio", GROUP="radio", MODE="0640"
 ACTION=="add", KERNEL=="bus/usb/*", OWNER="root", GROUP="usb", MODE="0660"
-ACTION=="add", KERNEL=="mtp_usb", OWNER="root", GROUP="mtp", MODE="0660"
+ACTION=="add", KERNEL=="mtp_usb", OWNER="root", GROUP="plugdev", MODE="0660"
 ACTION=="add", KERNEL=="usb_accessory", OWNER="root", GROUP="usb", MODE="0660"
 ACTION=="add", KERNEL=="tun", OWNER="system", GROUP="vpn", MODE="0660"
 ACTION=="add", KERNEL=="ts0710mux*", OWNER="radio", GROUP="radio", MODE="0640"
 ACTION=="add", KERNEL=="ppp", OWNER="radio", GROUP="vpn", MODE="0660"
 ACTION=="add", KERNEL=="dvb*", OWNER="root", GROUP="system", MODE="0660"
+rs.step_detector=true \
```
Comment (Flohack74) [inline ubuntu/70-hammerhead.rules:117]: Copy/Paste mistake?

File: ubuntu/apparmor.d/abstractions/base [hunk 2]
```diff
@@ -100,16 +100,16 @@
...
-  #  /android/odm/lib{,64}
-  /{,android/}vendor/lib{,64}/**           r,
-  /{,android/}vendor/lib{,64}/**.so        m,
-  /{,android/}system/lib{,64}/**           r,
-  /{,android/}system/lib{,64}/**.so        m,
-  /{,android/}system/vendor/lib{,64}/**    r,
-  /{,android/}system/vendor/lib{,64}/**.so m,
-  /{,android/}odm/lib{,64}/**    r,
-  /{,android/}odm/lib{,64}/**.so m,
+  #  /odm/lib{,32,64}
```
Comment (Flohack74) [inline ubuntu/apparmor.d/abstractions/base:103]: This is not needed, there are no folders called lib32

File: ubuntu/usb/mtp-server.conf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+description "MTP protocol server"
```
Comment (Flohack74) [inline ubuntu/usb/mtp-server.conf:1]: This is not needed normally, as the mtp-server startup is in the rootfs. If you made changes here, plz explain

File: init.hammerhead.rc [hunk 4]
```diff
@@ -390,10 +390,15 @@ service time_daemon /system/bin/time_daemon
     class late_start
     user root
     group root
-
+
```
Comment (Flohack74) [inline init.hammerhead.rc:393]: Remove whitespace change

File: device.mk [hunk 5]
```diff
@@ -246,7 +263,8 @@ PRODUCT_PACKAGES += \
 
 # Filesystem management tools
 PRODUCT_PACKAGES += \
-    e2fsck
+    e2fsck \
+    bluetoothd
```
Comment (Flohack74) [inline device.mk:273]: What is bluetoothd doing? We are working with Bluez in the UT rootfs, I don´t think this makes sense

File: device.mk [hunk 6]
```diff
@@ -141,8 +131,30 @@ PRODUCT_CHARACTERISTICS := nosdcard
...
+    miniafservice
+    
+MINIMEDIA_SENSORSERVER_DISABLE := 1
 
+# for off charging mode
 PRODUCT_PACKAGES += \
+    charger_res_images
+
+PRODUCT_PACKAGES += \
+    brcm_patchram_plus \
```
Comment (Flohack74) [inline device.mk:162]: Not needed anymore, patchram and hciattach are in a correctly patched version in teh rootfs

File: device.mk [hunk 7]
```diff
@@ -101,34 +102,23 @@ PRODUCT_COPY_FILES += \
...
-    libaudioflingerglue
 
 #Ubuntu Touch: USB port handling
-#PRODUCT_COPY_FILES += \
-#    $(LOCAL_PATH)/ubuntu/usb/setupusb:system/halium/usr/share/usbinit/setupusb \
-#    $(LOCAL_PATH)/ubuntu/usb/mtp-state.conf:system/halium/etc/init/mtp-state.conf \
-#    $(LOCAL_PATH)/ubuntu/usb/mtp-server.conf:system/halium/usr/share/upstart/sessions/mtp-server.conf
-
-# for off charging mode
-PRODUCT_PACKAGES += \
```
Comment (Flohack74) [inline device.mk:130]: Plase leave charger images inside, we want to have the same charging animation on as many devices as possible.

File: device.mk [hunk 8]
```diff
@@ -171,12 +183,17 @@ PRODUCT_PACKAGES += \
...
 
+# Audio
 PRODUCT_PACKAGES += \
     audio.primary.msm8974 \
     audio.a2dp.default \
-    audio.usb.default \
-    audio.r_submix.default \
-    libaudio-resampler \
+    audio.usb.defaultrs.step_detector=true \
+    ro.qti.sensors.step_counter=true \
```
Comment (Flohack74) [inline device.mk:197]: Why here again, you have it previously defined

File: proprietary-blobs.txt [hunk 9]
```diff
@@ -14,6 +14,12 @@
 
 # This file is generated by device/common/generate-blob-lists.sh - DO NOT EDIT
 
+-app/qcrilmsgtunnel/qcrilmsgtunnel.apk
```
Comment (Flohack74) [inline proprietary-blobs.txt:17]: I think apk and jar files are removed by halium setup script.

File: device.mk [hunk 10]
```diff
@@ -361,6 +385,10 @@ PRODUCT_PROPERTY_OVERRIDES += \
     persist.audio.fluence.voicecomm=true \
     persist.audio.fluence.voicerec=false \
     persist.audio.fluence.speaker=false
+
```
Comment (Flohack74) [inline device.mk:388]: Why move this here, that just creates noise in the PR. The part after the override files was already nicely in place.

File: device.mk (hunk 7 above)
Comment (rubencarneiro) [reply device.mk:130]: its the same

File: proprietary-blobs.txt (hunk 9 above)
Comment (rubencarneiro) [reply proprietary-blobs.txt:17]: Make no sense edit the file since the apk is copied

File: device.mk (hunk 6 above)
Comment (rubencarneiro) [reply device.mk:162]: Make all packages avaible to build if we use the prebuilts one. the packages are marked to build on the device tree, we should let it be, for new developers know why we build this and why is needed.

File: device.mk (hunk 5 above)
Comment (rubencarneiro) [reply device.mk:273]: Even if we work with bluez is good tho have the binary built in system.img for bluettohd, never know when you will needed a prebuilt binary for device, and it as no impact on the performance and wont cause any bugs.

File: ubuntu/apparmor.d/abstractions/base (hunk 2 above)
Comment (rubencarneiro) [reply ubuntu/apparmor.d/abstractions/base:103]: removing or adding as no effect, so wit will not respaw issues. and also is commented.

File: ubuntu/70-hammerhead.rules (hunk 1 above)
Comment (rubencarneiro) [reply ubuntu/70-hammerhead.rules:117]: yes typo error

File: device.mk (hunk 8 above)
Comment (rubencarneiro) [reply device.mk:197]: that a typo error
should be audio.usb.default i have to fix a

File: ubuntu/usb/mtp-server.conf (hunk 3 above)
Comment (rubencarneiro) [reply ubuntu/usb/mtp-server.conf:1]: this was removed on my device tree
#Ubuntu Touch: USB port handling
PRODUCT_COPY_FILES += \
    $(LOCAL_PATH)/ubuntu/usb/setupusb:system/halium/usr/share/usbinit/setupusb \
    $(LOCAL_PATH)/ubuntu/usb/mtp-state.conf:system/halium/etc/init/mtp-state.conf

File: ubuntu/70-hammerhead.rules (hunk 1 above)
Comment (rubencarneiro) [reply ubuntu/70-hammerhead.rules:117]: Fixed now

**Pull request closed without merging**